src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -170,59 +208,71 @@ public ComplaintResponse createComplaintResponseRepresentingLock(Complaint compl
      * @return complaintResponse of resolved complaint
      */
     public ComplaintResponse resolveComplaint(ComplaintResponse updatedComplaintResponse) {
-        if (updatedComplaintResponse.getId() == null) {
-            throw new IllegalArgumentException("The complaint response needs to have an id");
+        User user = this.userRepository.getUserWithGroupsAndAuthorities();
+
+        validateComplaintResponseId(updatedComplaintResponse.getId());
+        // TODO: make this retrieval redundant by proper fetching
+        ComplaintResponse complaintResponseFromDatabase = complaintResponseRepository.findByIdElseThrow(updatedComplaintResponse.getId());
+        validateComplaintResponseEmpty(complaintResponseFromDatabase);
+        // TODO: make this retrieval redundant by proper fetching
+        Complaint originalComplaint = complaintRepository.findByIdElseThrow(complaintResponseFromDatabase.getComplaint().getId());
+        validateOriginalComplaintNotAnswered(originalComplaint);
+
+        validateUserPermissionAndLockStatus(originalComplaint, complaintResponseFromDatabase, user);
```

Comment:
```suggestion
        validateUserPermissionAndLockStatus(originalComplaint, complaintResponseFromDatabase, user);
        validateComplaintResponseEmpty(complaintResponseFromDatabase);
        // TODO: make this retrieval redundant by proper fetching
        Complaint originalComplaint = complaintRepository.findByIdElseThrow(complaintResponseFromDatabase.getComplaint().getId());
        validateOriginalComplaintNotAnswered(originalComplaint);
```
This was what I meant by moving the access check to the top.

Reply:
This will not work since you define `originalComplaint` later but use it before.